Note on CI: the `lint (aks-node-controller)` check failure is **pre-existing on `main` and unrelated to this PR**.

- Error: `aks-node-controller/parser/helper.go:553:10: string "AzureChinaCloud" has 3 occurrences, make it a constant (goconst)`
- This PR touches only `parts/linux/cloud-init/artifacts/cse_config.sh` and its ShellSpec — **zero `aks-node-controller/` files** (and no Go at all).
- The flagged code already exists on `origin/main` at that line; the lint job runs with `only-new-issues: false`, so it reports the whole module's existing issues, not just this diff's.

All checks relevant to this change pass: shellcheck, shellspec, go-test, Pester, cue, dcgm-compatibility. Leaving the `goconst` fix out of scope for this hotfix.

--- a/spec/parts/linux/cloud-init/artifacts/cse_config_spec.sh
+++ b/spec/parts/linux/cloud-init/artifacts/cse_config_spec.sh
@@ -146,6 +146,19 @@ Describe 'cse_config.sh'
             The output should not include "STUB_TEARDOWN_CALLED"
             The status should be success
         End
+
+        It 'is a no-op on a non-Ubuntu OS even when a mismatched marker is present'
+            marker="$(mktemp)"
+            printf 'driver_kind=cuda\n' > "$marker"
+            GPU_DKMS_MARKER_FILE="$marker"
+            OS="MARINER"   # override the Ubuntu default set at the Describe level
+            NVIDIA_GPU_DRIVER_TYPE="grid"
+            When call cleanUpGridNodeCudaPrebake
+            The output should not include "STUB_TEARDOWN_CALLED"
+            The status should be success
+            OS="$UBUNTU_OS_NAME"   # restore for any subsequent examples
+            rm -f "$marker"
+        End
     End
 
     Describe 'configureAzureJson'

